### PR TITLE
Fix kdump service status wrong check and summary.log always is covered

### PR DESCRIPTION
**Pull request by HiRoySoft**
In this Pull requests, it includes five fixes to improve kdump test stability 

1. Source utils.sh in kdump_execute.sh and kdump_results,sh, so that it is easy to operate every Distribution's detailed Versions with global variable DISTRO

2. Fix summary.log got from VM always is covered by next case, now get every case's three stages log in Kdump.ps1

3. Stop the second VM after vmcore checking fails in Kdump.ps1

4. Add kdump_nfs_config.sh into "files" label in Kdump_Test.xml, or Crash_nfs will be fail as no kdump_nfs_config.sh is pushed

5. Add snapshot for every kdump case, as /etc/kdump.config will save nfs=nfs_ip:/mnt for other cases after Crash_nfs case execution is done

### CODE REVIEW

File: WS2012R2/lisa/xml/Kdump_Tests.xml [hunk 1]
```diff
@@ -54,11 +54,12 @@
         <test>
             <testName>Crash_single_core</testName>
             <setupScript>
+                <file>setupscripts\RevertSnapshot.ps1</file>
                 <file>setupscripts\ChangeCPU.ps1</file>
                 <file>setupScripts\SetVMMemory.ps1</file>
             </setupScript>
             <testScript>setupScripts\kdump.ps1</testScript>
-            <files>remote-scripts/ica/utils.sh,remote-scripts/ica/kdump_config.sh,remote-scripts/ica/kdump_results.sh,remote-scripts/ica/kdump_execute.sh</files>
+            <files>remote-scripts/ica/utils.sh,remote-scripts/ica/kdump_config.sh,remote-scripts/ica/kdump_results.sh,remote-scripts/ica/kdump_execute.sh,remote-scripts/ica/kdump_nfs_config.sh</files>
```
Comment (chvalean) [inline WS2012R2/lisa/xml/Kdump_Tests.xml]: pushing kdump_nfs_config.sh is not required for each test case, unless you use for all test cases a NFS target.
But that implies at least the other param to be added to each test case.

File: WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh [hunk 2]
```diff
@@ -30,69 +46,64 @@ sys_kexec_crash=/sys/kernel/kexec_crash_loaded
 #
 # Functions definitions
 #
-LogMsg()
-{
-    # To add the time-stamp to the log file
-    echo `date "+%a %b %d %T %Y"` ": ${1}"
-}
-
 UpdateTestState()
```
Comment (adriansuhov) [inline WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh]: utils.sh has this already. You can remove this function

File: WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh [hunk 3]
```diff
@@ -192,17 +207,21 @@ ConfigureNMI()
 #
```
Comment (adriansuhov) [inline WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh]: Please source utils.sh and use SetTestStateAborted function from there

File: WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh [hunk 4]
```diff
@@ -21,6 +21,22 @@
...
+	echo "Error: unable to source utils.sh!"
+	exit 1
+}
+
+#
+# Source constants file and initialize most common variables
+#
+UtilsInit
+
 ICA_TESTABORTED="TestAborted"
```
Comment (adriansuhov) [inline WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh]: You can remove this line and rely on the function provided by utils.sh

File: WS2012R2/lisa/remote-scripts/ica/Kdump_Results.sh [hunk 5]
```diff
@@ -19,32 +19,27 @@
...
+	echo "Error: unable to source utils.sh!"
+	exit 1
 }
 
+#
+# Source constants file and initialize most common variables
+#
+UtilsInit
+
+ICA_TESTFAILED="TestFailed"
```
Comment (adriansuhov) [inline WS2012R2/lisa/remote-scripts/ica/Kdump_Results.sh]: Same as Kdump_Execute.sh. Please remove all variables and functions, and use SetTestState functions from utils.sh

File: WS2012R2/lisa/xml/Kdump_Tests.xml (hunk 1 above)
Comment (HiRoySoft) [inline WS2012R2/lisa/xml/Kdump_Tests.xml]: Yes, after submitted pull requests, i noted the mistake.
Just left it for Crash_nfs case

File: WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh (hunk 4 above)
Comment (HiRoySoft) [inline WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh]: Yes, will remove them and check other functions from utils.sh

Thanks.

File: WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh (hunk 2 above)
Comment (HiRoySoft) [inline WS2012R2/lisa/remote-scripts/ica/Kdump_Execute.sh]: OK, the same operation likes LogMsg

File: WS2012R2/lisa/remote-scripts/ica/Kdump_Results.sh (hunk 5 above)
Comment (HiRoySoft) [inline WS2012R2/lisa/remote-scripts/ica/Kdump_Results.sh]: Got it

Comment (HiRoySoft) [inline WS2012R2/lisa/remote-scripts/ica/Kdump_Results.sh]: Will operate them in kdump_config.sh, kdump_execute.sh kdump_resutls.sh.

That's great!

**Pull request merged**